docs/contributing.md

```diff
@@ -56,23 +33,23 @@ Below are a few different queries you can use to find appropriate issues to work
 Patches should be submitted as [pull requests](https://help.github.com/articles/about-pull-requests/) (PRs).
 
 Before submitting a PR:
-- Run `cargo fmt` to ensure your Rust code is correctly formatted.
-  - If you have modified any Swift code, also run `swiftformat --swiftversion 4` on the modified code.
-- Your code must run and pass all the automated tests before you submit your PR for review.
-  - The simplest way to confirm this is to run `cargo all_tests` which uses the `./automation/all_tests.sh` script, that runs all test suites
-    and linters for Rust, Kotlin and Swift code.
-  - "Work in progress" pull requests are welcome, but should be clearly labeled as such and should not be merged until all tests pass and the code has been reviewed.
-- Your patch should include new tests that cover your changes, or be accompanied by explanation for why it doesn't need any. It is your
-  and your reviewer's responsibility to ensure your patch includes adequate tests.
+- Your patch should include new tests that cover your changes, or be accompanied by explanation for why it doesn't need any. It is your and your reviewer's responsibility to ensure your patch includes adequate tests.
   - Consult the [testing guide](./howtos/testing-a-rust-component.md) for some tips on writing effective tests.
+- Your code should pass all the automated tests before you submit your PR for review.
+  - The simplest way to confirm this is to run `cargo all_tests`, which uses the `./automation/all_tests.sh` script, that runs all test suites and linters for Rust, Kotlin and Swift code.
+    - **Note:** You should avoid running all automated tests to incrementally validate your changes as it can be a long-running process. Instead run a reasonable subset of all tests that will exercise your changes and then run `cargo all_tests` to validate your entire patch before submitting your PR.
```

Comment:
I think the "should" here is a bit too strong - something like "you might choose to" might make it clearer that you are just suggesting an optimization.